src/qt/rpcconsole.cpp

```diff
@@ -616,8 +616,16 @@ void RPCConsole::on_lineEdit_returnPressed()
 
     if(!cmd.isEmpty())
     {
-        message(CMD_REQUEST, cmd);
         Q_EMIT cmdRequest(cmd);
+        // Remove passphrases from history
+        QStringList cmdList = cmd.split(' ');
```

Comment:
Sure. What I'm afraid of is that in the future someone is bound to add a command for which this is not true. The handling per command is a bit brittle.

So I'd prefer these commands to not be added to history in the first place.